File: xamarin-Mvvmcross/PropertyCross.Core/Domain/Model/IJsonPropertySearch.cs
using System;

namespace PropertyCross.Core.Domain.Model
{
  public interface IJsonPropertySearch
  {
    void FindProperties(string location, int pageNumber, Action<string> callback, Action<Exception> error);

    void FindProperties(double latitude, double longitude, int pageNumber, Action<string> callback, Action<Exception> error);
  }
}

File: native/android/PropertyCross/app/src/main/java/com/propertycross/android/model/IJsonPropertySearch.java
package com.propertycross.android.model;

import com.propertycross.android.events.Callback;

public interface IJsonPropertySearch {
	
	public void findProperties(String searchText, int pageNumber,
			Callback<String> complete, Callback<Exception> error);
	
	public void findProperties(double latitude, double longitude, int pageNumber,
			Callback<String> complete, Callback<Exception> error);
}